packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker-task-base-types.ts

```diff
@@ -626,6 +626,181 @@ export enum AssembleWith {
 
 }
 
+/**
+ * Specifies the containers in the inference pipeline.
+ *
+ * @experimental
+ */
+export interface ModelContainer {
+
+    /**
+     * This parameter is ignored for models that contain only a PrimaryContainer
+     */
+    readonly containerHostname?: string
+
+    /**
+     * The environment variables to set in the Docker container.
+     * Each key and value in the Environment string to string map can have length of up to 1024.
+     * We support up to 16 entries in the map.
+     */
+    readonly environment?: {[key: string]: string}
+
+    /**
+     * The Amazon EC2 Container Registry (Amazon ECR) path where inference code is stored.
+     */
+    readonly image: DockerImage
+
+    /**
+     * Whether the container hosts a single model or multiple models.
+     */
+    readonly mode?: ModelContainerMode
+
+    /**
+     * The S3 path where the model artifacts, which result from model training, are stored.
+     * This path must point to a single gzip compressed tar archive (.tar.gz suffix).
+     * The S3 path is required for Amazon SageMaker built-in algorithms, but not if you use your own algorithms.
+     */
+    readonly modelDataUrl?: string
+
+    /**
+     * The name or Amazon Resource Name (ARN) of the model package to use to create the model.
+     */
+    readonly modelPackageName?: string
+
+}
+
+/**
+ * Whether the container hosts a single model or multiple models
+ *
+ * @experimental
+ */
+export enum ModelContainerMode {
+    SINGLEMODEL = 'SingleModel',
+    MULTIMODEL = 'MultiModel'
+}
+
+/**
+ * Specifies the containers in the inference pipeline.
+ *
+ * @experimental
+ */
+export interface ProductionVariants {
```

Comment:
This interface should be renamed to ProductionVariant instead of ProductionVariants, as it defines one ProductionVariant. 
Also defining this interface here will introduce code duplication when  [PR for L2 sagemaker constructs](https://github.com/aws/aws-cdk/pull/6107) is merged and [ProductionVariant](https://github.com/aws/aws-cdk/pull/6107/files#diff-634525d1dab98e727f4d5cd66fb2fb46R70) gets into the package. After mentioned PR being live, created step function tasks should use the real defined ProductionVariant in sagemaker instead of introducing another one.